metainsuranceorg.py

```diff
@@ -330,4 +366,55 @@ def adjust_dividend(self, time):
         
     def adjust_capacity_target(self, time):
         assert False, "Method not implemented. adjust_dividend method should be implemented in inheriting classes"
+
+    def risks_reinrisks_organizer(self, new_risks):  #This method organizes the new risks received by the insurer (or reinsurer)
+
+        risks_per_categ = [[] for x in range(self.simulation_parameters["no_categories"])]      #This method organizes the new risks received by the insurer (or reinsurer) by category in the nested list "risks_per_categ".
+        number_risks_categ = [[] for x in range(self.simulation_parameters["no_categories"])]   #This method also counts the new risks received by the insurer (or reinsurer) by category in the list "number_risks_categ".
+
+        for categ_id in range(self.simulation_parameters["no_categories"]):
+            risks_per_categ[categ_id] = [risk for risk in new_risks if risk["category"] == categ_id]
+            number_risks_categ[categ_id] = len(risks_per_categ[categ_id])
+
+        return risks_per_categ, number_risks_categ   #The method returns both risks_per_categ and risks_per_categ.
+
+    def balanced_portfolio(self, risk, cash_left_by_categ, var_per_risk):
+
+        cash_reserved_by_categ = np.zeros(self.simulation_parameters["no_categories"])
+
+        for i in range(len(cash_left_by_categ)):
+            cash_reserved_by_categ[i] = self.cash - cash_left_by_categ[i]     #Here it is computed the cash already reserved by category
+
+        std_pre = cash_reserved_by_categ.std()                                #Here it is computed the standard deviation of the cash reserved by category
+
+        if risk.get("insurancetype")=='excess-of-loss':
+            cash_reserved_by_categ_store = np.copy(cash_reserved_by_categ)
+            percentage_value_at_risk = self.riskmodel.getPPF(categ_id=risk["category"], tailSize=self.riskmodel.var_tail_prob)
+            expected_damage = percentage_value_at_risk * risk["value"] * risk["risk_factor"] \
+                              * self.riskmodel.inaccuracy[risk["category"]]
+            expected_claim = min(expected_damage, risk["value"] * risk["excess_fraction"]) - risk["value"] * risk["deductible_fraction"]
+
+            # record liquidity requirement and apply margin of safety for liquidity requirement
+
+            cash_reserved_by_categ_store[risk["category"]] += expected_claim * self.riskmodel.margin_of_safety  #Here it is computed how the cash reserved by category would change if the new reinsurance risk was accepted
+
+        else:
+            cash_reserved_by_categ_store = np.copy(cash_reserved_by_categ)
+            cash_reserved_by_categ_store[risk["category"]] = cash_reserved_by_categ_store[risk["category"]] + var_per_risk[risk["category"]] #Here it is computed how the cash reserved by category would change if the new insurance risk was accepted
```

Comment:
metainsuranceorg.py, line 403: Why not use `+=` increment operator, especially with variable names as long as these, that would improve readability.

Reply:
Done.